engine/model.py

```diff
@@ -341,6 +353,39 @@ def _prevent_oob(self) -> None:
 
         self.set_player_speed(new_vx, new_vy)
 
+    def handle_collisions_with_solids(self) -> None:
+        """Prevents the player from moving through solid objects."""
+        if self.scene is None:
+            raise SceneNotInitialized()
+
+        new_x = self.player_sprite.center_x + self.player_sprite.change_x
+        new_y = self.player_sprite.center_y + self.player_sprite.change_y
```

Comment:
There's a lot of duplicated code here, that's typically a code smell. Why not just merge this code with `_prevent_oob()`?